src/main/java/org/dcsa/bkg/service/impl/BKGServiceImpl.java

```diff
@@ -119,8 +115,11 @@ public Mono<BookingResponseTO> createBooking(final BookingTO bookingRequest) {
             booking ->
                 createDeepObjectsForBooking(bookingRequest, booking)
                     .flatMap(
-                        bTO ->
-                            createShipmentEventFromBookingTO(booking.getId(), bTO).thenReturn(bTO)))
+                        bookingTO ->
+                            this.createShipmentEvent(
+                                    shipmentEventMapper.shipmentEventFromBookingTO(
+                                        bookingTO, booking.getId(), null))
+                                .thenReturn(bookingTO)))
         .flatMap(bTO -> Mono.just(bookingMapper.dtoToBookingResponseTO(bTO)));
```

Comment:
This line (which is not new) could be simplified to
```suggestion
         .map(bookingMapper::dtoToBookingResponseTO);
```

Reply:
Resolving as it is not a blocker.

Reply:
changed